[IMP] l10n_ro: Minor changes for l10n_ro_saft

antoine162 opened the pull request:
For the l10n_ro_saft module that generates the D.406 declaration, we needed to:
- create a new export tax specifically for services (which should be reported separately from goods); and
- make sure the Bank, Outstanding Receipts and Outstanding Payments accounts are created with codes 5121xx, and the Cash account with code 5311xx, because codes 5120 and 5130 are not available in the official CoA and were therefore causing validation errors in the SAF-T export.
- because the CUI number (found in the company_registry field) for partners is required for the SAF-T export, and it is substantially the same as the VAT number, re-use logic from l10n_be to automatically fill in the company_registry if the VAT exists.

Enterprise PR: https://github.com/odoo/enterprise/pull/43241
Documentation PR: https://github.com/odoo/documentation/pull/4916

task-id: 3172198

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/126563).

antoine162 commented on addons/l10n_ro/data/account_tax_data.xml:
```diff
@@ -1145,7 +1145,30 @@
+    </record>
+    <record id="tvati_extras" model="account.tax.template">
+        <field name="chart_template_id" ref="ro_chart_template"/>
+        <field name="sequence">35</field>
```
Thanks so much!

jco-odoo commented on addons/l10n_ro/models/res_partner.py:
```diff
@@ -18,3 +18,13 @@ def _commercial_fields(self):
+        super()._compute_company_registry()
+        for partner in self.filtered(lambda p: p.country_id.code == 'RO' and p.vat):
+            vat_country, vat_number = self._split_vat(partner.vat)
+            if vat_country == 'ro' and self.simple_vat_check(vat_country, vat_number):
```
@antoine162 Please check https://github.com/odoo/odoo/pull/125058#issuecomment-1623836753

antoine162 replied:
Thanks. I didn't think about the case where the user puts the VAT number without the 'RO' prefix. Fixed now in the same way as Brice's PR.

jco-odoo commented on addons/l10n_ro/data/account_tax_data.xml:
```diff
@@ -1145,7 +1145,30 @@
         <field name="chart_template_id" ref="ro_chart_template"/>
         <field name="sequence">35</field>
-        <field name="name">TVA Export</field>
+        <field name="name">TVA Export Bunuri</field>
```
Normally we add this to have it in the fiscal positions...

antoine162 replied:
Yes :sweat_smile:

jco-odoo replied:
still missing ?

antoine162 replied:
Thanks.

Levizar commented on addons/l10n_ro/models/account_move.py:
```diff
@@ -0,0 +1,14 @@
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_ro_is_self_invoice = fields.Boolean(
+        string='Is self invoice?',
+        help='Check this box to indicate that this bill is a self-invoice made to register '
+             'a VAT event in the absence of an invoice received from a supplier.',
+    )
```
I think I discussed those kind of invoice with TSB a while ago.
Don't we have some standard stuff for that?

antoine162 replied:
Do we?

antoine162 replied:
I guess that in the end the answer from Benj was that we don't have any standard thing for that...

jco-odoo replied:
Ideally, we would see if we have some tax related to certain lines in the tax report, but this might be the simplest for now.

jco-odoo replied:
@antoine162 @Levizar  I am not against having the field to go forward right now, but you can not add it in stable in the module in 16.  An option is to simply move it to l10n_ro_saft for now.

antoine162 commented:
Hey @jco-odoo thank you so much for this last review, I won't bother you since you're on holiday, but FYI I have added the missing fiscal positions and moved the is_self_invoice field to l10n_ro_saft. Once this is merged in master, I'll move it back to the l10n_ro module (probably necessary for the Romanian EDI).

Levizar reviewed (approved):
@robodoo r+

robodoo commented:
@antoine162 @Levizar linked pull request(s) odoo/documentation#4916, odoo/enterprise#43241 not ready. Linked PRs are not staged until all of them are ready.

Levizar reviewed (approved):
@robodoo r+
